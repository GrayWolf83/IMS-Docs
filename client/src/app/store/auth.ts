import { setLoadingError } from './error'
import { IUser } from './../../../../server/types/User'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import authService from '../services/auth.service'
import localStorageService from '../services/localStorage.service'

type authState = {
	currentUser: IUser | null
	isLoading: boolean
}

const initialState: authState = {
	currentUser: null,
	isLoading: false,
}

const authSlice = createSlice({
	name: 'auth',
	initialState,
	reducers: {
		authLoginSuccess(state, action: PayloadAction<IUser>) {
			state.currentUser = action.payload
		},
		authLogout(state) {
			state.currentUser = null
		},
		authLoadingStart(state) {
			state.isLoading = true
		},
		authLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const { authLoginSuccess, authLoadingStart, authLoadingEnd, authLogout } =
	authSlice.actions

export const login = (data: FormData) => async (dispatch: AppDispatch) => {
	dispatch(authLoadingStart())

	try {
		const payload = await authService.login(data)
		dispatch(authLoginSuccess(payload.user))

		localStorageService.setTokens({
			...payload,
		})
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(authLoadingEnd())
	}
}

export const autologin = () => async (dispatch: AppDispatch) => {
	dispatch(authLoadingStart())
	try {
		const payload = await authService.refresh()

		if (!payload?.refreshToken || !payload?.accessToken) {
			localStorageService.removeAuthData()
		} else {
			dispatch(authLoginSuccess(payload.user))

			localStorageService.setTokens({
				...payload,
			})
		}
	} catch (error: any) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(authLoadingEnd())
	}
}

export const logout = () => async (dispatch: AppDispatch) => {
	localStorageService.removeAuthData()
	dispatch(authLogout())
}

export const getCurrentUser = () => (state: RootState) => {
	return state.auth.currentUser
}

export const getAuthLoadingStatus = () => (state: RootState) => {
	return state.auth.isLoading
}

export const getCurrentUserIsManage = () => (state: RootState) => {
	return Boolean(state.auth.currentUser?.role === 'Manage')
}

export default authSlice.reducer
