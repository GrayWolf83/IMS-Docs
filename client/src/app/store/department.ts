import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { IDepartment } from '../types/department'
import departmentService from '../services/department.service'

type state = {
	entities: IDepartment[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const departmentSlice = createSlice({
	name: 'departments',
	initialState,
	reducers: {
		departmentsLoaded(state, action: PayloadAction<IDepartment[]>) {
			state.entities = action.payload
		},
		departmentAdded(state, action: PayloadAction<IDepartment>) {
			state.entities = [action.payload, ...state.entities]
		},
		departmentEdited(state, action: PayloadAction<IDepartment>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		departmentDeleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		departmentLoadingStart(state) {
			state.isLoading = true
		},
		departmentLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const {
	departmentsLoaded,
	departmentAdded,
	departmentEdited,
	departmentDeleted,
	departmentLoadingStart,
	departmentLoadingEnd,
} = departmentSlice.actions

export const loadDepartmentList = () => async (dispatch: AppDispatch) => {
	dispatch(departmentLoadingStart())

	try {
		const payload = await departmentService.getList()
		dispatch(departmentsLoaded(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(departmentLoadingEnd())
	}
}

export const addDepartment =
	(data: { name: string; fullName: string; index: number }) =>
	async (dispatch: AppDispatch) => {
		dispatch(departmentLoadingStart())
		try {
			const payload: IDepartment = await departmentService.add(data)

			dispatch(departmentAdded(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(departmentLoadingEnd())
		}
	}

export const editDepartment =
	(data: IDepartment) => async (dispatch: AppDispatch) => {
		dispatch(departmentLoadingStart())
		try {
			const payload: IDepartment = await departmentService.edit(data)

			dispatch(departmentEdited(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(departmentLoadingEnd())
		}
	}

export const deleteDepartment =
	(data: string) => async (dispatch: AppDispatch) => {
		dispatch(departmentLoadingStart())
		try {
			const payload: string = await departmentService.delete(data)

			dispatch(departmentDeleted(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(departmentLoadingEnd())
		}
	}

export const getDepartmentsList = () => (state: RootState) => {
	return state.departments.entities
		.filter((item) => item.index)
		.sort((a, b) => (a.index > b.index ? 1 : -1))
}

export const getDepartmentById = (id: string) => (state: RootState) => {
	return state.departments.entities.find((item) => item.id === id)
}

export default departmentSlice.reducer
