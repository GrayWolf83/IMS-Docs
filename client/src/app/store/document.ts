import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { IDocument, INewDocument } from '../types/document'
import documentService from '../services/document.service'

type state = {
	entities: IDocument[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const documentSlice = createSlice({
	name: 'documents',
	initialState,
	reducers: {
		loaded(state, action: PayloadAction<IDocument[]>) {
			state.entities = action.payload
		},
		added(state, action: PayloadAction<IDocument>) {
			state.entities = [action.payload, ...state.entities]
		},
		edited(state, action: PayloadAction<IDocument>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		deleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		loadingStart(state) {
			state.isLoading = true
		},
		loadingEnd(state) {
			state.isLoading = false
		},
	},
})

const { loaded, added, edited, deleted, loadingStart, loadingEnd } =
	documentSlice.actions

export const loadDocumentsList =
	(systemId: string) => async (dispatch: AppDispatch) => {
		dispatch(loadingStart())

		try {
			const payload: IDocument[] = await documentService.getList(systemId)
			dispatch(loaded(payload))
		} catch (error) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(loadingEnd())
		}
	}

export const addDocument =
	(data: INewDocument) => async (dispatch: AppDispatch) => {
		dispatch(loadingStart())
		try {
			const payload: IDocument = await documentService.add(data)

			dispatch(added(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(loadingEnd())
		}
	}

export const editDocument =
	(data: IDocument) => async (dispatch: AppDispatch) => {
		dispatch(loadingStart())
		try {
			const payload: IDocument = await documentService.edit(data)

			dispatch(edited(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(loadingEnd())
		}
	}

export const deleteDocument =
	(data: string) => async (dispatch: AppDispatch) => {
		dispatch(loadingStart())
		try {
			const payload: string = await documentService.delete(data)

			dispatch(deleted(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(loadingEnd())
		}
	}

export const getDocumentsList = () => (state: RootState) => {
	return state.documents.entities
}

export const getDocumentById = (id: string) => (state: RootState) => {
	return state.documents.entities.find((item) => item.id === id)
}

export default documentSlice.reducer
