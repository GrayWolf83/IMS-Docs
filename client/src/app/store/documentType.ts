import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { IDocumentType, INewDocumentType } from '../types/documentType'
import documentTypeService from '../services/documentType.service'

type state = {
	entities: IDocumentType[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const documentTypeSlice = createSlice({
	name: 'documentTypes',
	initialState,
	reducers: {
		documentTypesLoaded(state, action: PayloadAction<IDocumentType[]>) {
			state.entities = action.payload
		},
		documentTypeAdded(state, action: PayloadAction<IDocumentType>) {
			state.entities = [action.payload, ...state.entities]
		},
		documentTypeEdited(state, action: PayloadAction<IDocumentType>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		documentTypeDeleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		documentTypeLoadingStart(state) {
			state.isLoading = true
		},
		documentTypeLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const {
	documentTypesLoaded,
	documentTypeAdded,
	documentTypeEdited,
	documentTypeDeleted,
	documentTypeLoadingStart,
	documentTypeLoadingEnd,
} = documentTypeSlice.actions

export const loadDocumentTypesList = () => async (dispatch: AppDispatch) => {
	dispatch(documentTypeLoadingStart())

	try {
		const payload: IDocumentType[] = await documentTypeService.getList()
		dispatch(documentTypesLoaded(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(documentTypeLoadingEnd())
	}
}

export const addDocumentType =
	(data: INewDocumentType) => async (dispatch: AppDispatch) => {
		dispatch(documentTypeLoadingStart())
		try {
			const payload: IDocumentType = await documentTypeService.add(data)

			dispatch(documentTypeAdded(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(documentTypeLoadingEnd())
		}
	}

export const editDocumentType =
	(data: IDocumentType) => async (dispatch: AppDispatch) => {
		dispatch(documentTypeLoadingStart())
		try {
			const payload: IDocumentType = await documentTypeService.edit(data)

			dispatch(documentTypeEdited(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(documentTypeLoadingEnd())
		}
	}

export const deleteDocumentType =
	(data: string) => async (dispatch: AppDispatch) => {
		dispatch(documentTypeLoadingStart())
		try {
			const payload: string = await documentTypeService.delete(data)

			dispatch(documentTypeDeleted(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(documentTypeLoadingEnd())
		}
	}

export const getDocumentTypesList = () => (state: RootState) => {
	return state.documentTypes.entities
}

export const getDocumentTypeById = (id: string) => (state: RootState) => {
	return state.documentTypes.entities.find((item) => item.id === id)
}

export default documentTypeSlice.reducer
