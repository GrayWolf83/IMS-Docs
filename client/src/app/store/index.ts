import { configureStore } from '@reduxjs/toolkit'
import errorReducer from './error'
import authReducer from './auth'
import systemReducer from './system'
import departmentReducer from './department'
import positionReducer from './position'
import userReducer from './user'
import documentTypesReducer from './documentType'

const store = configureStore({
	reducer: {
		errors: errorReducer,
		auth: authReducer,
		systems: systemReducer,
		departments: departmentReducer,
		positions: positionReducer,
		users: userReducer,
		documentTypes: documentTypesReducer,
	},
})

export default store

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
