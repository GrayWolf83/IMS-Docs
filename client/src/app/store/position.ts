import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { IPosition } from '../types/position'
import positionService from '../services/position.service'

type state = {
	entities: IPosition[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const positionSlice = createSlice({
	name: 'positions',
	initialState,
	reducers: {
		positionsLoaded(state, action: PayloadAction<IPosition[]>) {
			state.entities = action.payload
		},
		positionAdded(state, action: PayloadAction<IPosition>) {
			state.entities = [action.payload, ...state.entities]
		},
		positionEdited(state, action: PayloadAction<IPosition>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		positionDeleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		positionLoadingStart(state) {
			state.isLoading = true
		},
		positionLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const {
	positionsLoaded,
	positionAdded,
	positionEdited,
	positionDeleted,
	positionLoadingStart,
	positionLoadingEnd,
} = positionSlice.actions

export const loadPositionsList = () => async (dispatch: AppDispatch) => {
	dispatch(positionLoadingStart())

	try {
		const payload: IPosition[] = await positionService.getList()
		dispatch(positionsLoaded(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(positionLoadingEnd())
	}
}

export const addPosition =
	(data: { name: string; index: number }) =>
	async (dispatch: AppDispatch) => {
		dispatch(positionLoadingStart())
		try {
			const payload: IPosition = await positionService.add(data)

			dispatch(positionAdded(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(positionLoadingEnd())
		}
	}

export const editPosition =
	(data: IPosition) => async (dispatch: AppDispatch) => {
		dispatch(positionLoadingStart())
		try {
			const payload: IPosition = await positionService.edit(data)

			dispatch(positionEdited(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(positionLoadingEnd())
		}
	}

export const deletePosition =
	(data: string) => async (dispatch: AppDispatch) => {
		dispatch(positionLoadingStart())
		try {
			const payload: string = await positionService.delete(data)

			dispatch(positionDeleted(payload))
		} catch (error: any) {
			dispatch(setLoadingError(error))
		} finally {
			dispatch(positionLoadingEnd())
		}
	}

export const getPositionsList = () => (state: RootState) => {
	return state.positions.entities
		.filter((item) => item.index)
		.sort((a, b) => (a.index > b.index ? 1 : -1))
}

export const getPositionById = (id: string) => (state: RootState) => {
	return state.positions.entities.find((item) => item.id === id)
}

export default positionSlice.reducer
