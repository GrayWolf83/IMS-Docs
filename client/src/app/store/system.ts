import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { ISystem } from '../types/system'
import systemService from '../services/system.service'

type state = {
	entities: ISystem[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const systemSlice = createSlice({
	name: 'systems',
	initialState,
	reducers: {
		systemsLoaded(state, action: PayloadAction<ISystem[]>) {
			state.entities = action.payload
		},
		systemAdded(state, action: PayloadAction<ISystem>) {
			state.entities = [action.payload, ...state.entities]
		},
		systemEdited(state, action: PayloadAction<ISystem>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		systemDeleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		systemLoadingStart(state) {
			state.isLoading = true
		},
		systemLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const {
	systemsLoaded,
	systemAdded,
	systemEdited,
	systemDeleted,
	systemLoadingStart,
	systemLoadingEnd,
} = systemSlice.actions

export const loadSystemsList = () => async (dispatch: AppDispatch) => {
	dispatch(systemLoadingStart())

	try {
		const payload = await systemService.getList()
		dispatch(systemsLoaded(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(systemLoadingEnd())
	}
}

export const addSystem = (data: ISystem) => async (dispatch: AppDispatch) => {
	dispatch(systemLoadingStart())
	try {
		const payload: ISystem = await systemService.add(data)

		dispatch(systemAdded(payload))
	} catch (error: any) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(systemLoadingEnd())
	}
}

export const editSystem = (data: ISystem) => async (dispatch: AppDispatch) => {
	dispatch(systemLoadingStart())
	try {
		const payload: ISystem = await systemService.edit(data)

		dispatch(systemEdited(payload))
	} catch (error: any) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(systemLoadingEnd())
	}
}

export const deleteSystem = (data: string) => async (dispatch: AppDispatch) => {
	dispatch(systemLoadingStart())
	try {
		const payload: string = await systemService.delete(data)

		dispatch(systemDeleted(payload))
	} catch (error: any) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(systemLoadingEnd())
	}
}

export const getSystemsList = () => (state: RootState) => {
	return state.systems.entities
}

export default systemSlice.reducer
