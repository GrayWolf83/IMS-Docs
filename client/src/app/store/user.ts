import { INewUser } from './../types/user'
import { setLoadingError } from './error'
import { AppDispatch, RootState } from './index'
import { createSlice, PayloadAction } from '@reduxjs/toolkit'
import { IUser } from '../types/user'
import userService from '../services/user.service'

type state = {
	entities: IUser[]
	isLoading: boolean
}

const initialState: state = {
	entities: [],
	isLoading: false,
}

const userSlice = createSlice({
	name: 'users',
	initialState,
	reducers: {
		usersLoaded(state, action: PayloadAction<IUser[]>) {
			state.entities = action.payload
		},
		userAdded(state, action: PayloadAction<IUser>) {
			state.entities = [action.payload, ...state.entities]
		},
		userEdited(state, action: PayloadAction<IUser>) {
			state.entities = state.entities.map((item) => {
				return item.id === action.payload.id ? action.payload : item
			})
		},
		userDeleted(state, action: PayloadAction<string>) {
			state.entities = state.entities.filter(
				(item) => item.id !== action.payload,
			)
		},
		userLoadingStart(state) {
			state.isLoading = true
		},
		userLoadingEnd(state) {
			state.isLoading = false
		},
	},
})

const {
	usersLoaded,
	userAdded,
	userEdited,
	userDeleted,
	userLoadingStart,
	userLoadingEnd,
} = userSlice.actions

export const loadUsersList = () => async (dispatch: AppDispatch) => {
	dispatch(userLoadingStart())
	try {
		const payload: IUser[] = await userService.getList()
		dispatch(usersLoaded(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(userLoadingEnd())
	}
}

export const addUser = (data: INewUser) => async (dispatch: AppDispatch) => {
	dispatch(userLoadingStart())
	try {
		const payload: IUser = await userService.add(data)
		dispatch(userAdded(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(userLoadingEnd())
	}
}

export const editUser = (data: IUser) => async (dispatch: AppDispatch) => {
	dispatch(userLoadingStart())
	try {
		const payload: IUser = await userService.edit(data)
		dispatch(userEdited(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(userLoadingEnd())
	}
}

export const deleteUser = (data: string) => async (dispatch: AppDispatch) => {
	dispatch(userLoadingStart())
	try {
		const payload: string = await userService.delete(data)
		dispatch(userDeleted(payload))
	} catch (error) {
		dispatch(setLoadingError(error))
	} finally {
		dispatch(userLoadingEnd())
	}
}

export const getUsersList = () => (state: RootState) => {
	return state.users.entities
}

export const getUserById = (id: string) => (state: RootState) => {
	return state.users.entities.find((user) => user.id === id)
}

export default userSlice.reducer
