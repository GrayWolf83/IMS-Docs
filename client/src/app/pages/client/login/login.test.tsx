import React from 'react'
import { render, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import Login from './login'
import { Provider } from 'react-redux'
import store from '../../../store'

describe('Login page', () => {
	test('Login render', () => {
		render(
			<Provider store={store}>
				<BrowserRouter>
					<Login />
				</BrowserRouter>
			</Provider>,
		)

		const login = screen.getByTestId('Login')
		expect(login).toBeInTheDocument()
	})

	test('Login snapshot', () => {
		expect(
			render(
				<Provider store={store}>
					<BrowserRouter>
						<Login />
					</BrowserRouter>
				</Provider>,
			),
		).toMatchSnapshot()
	})
})
