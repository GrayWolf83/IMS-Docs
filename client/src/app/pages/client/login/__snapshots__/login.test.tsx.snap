// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page Login snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="css-0"
        data-testid="Header"
      >
        <div
          class="chakra-container css-0"
        >
          <div
            class="css-0"
          >
            <a
              href="/"
            >
              <h2
                class="chakra-heading css-0"
              >
                IMS-Docs
              </h2>
            </a>
            <div
              class="css-0"
            >
              <button
                aria-label="Switch to dark mode"
                class="chakra-button css-0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </header>
      <div
        class="css-0"
        data-testid="Login"
      >
        <h2
          class="chakra-heading css-0"
        >
          Авторизация
        </h2>
        <div
          class="css-0"
          data-testid="FormComponent"
        >
          <form
            enctype="multipart/form-data"
            style="width: 100%;"
          >
            <div
              class="chakra-form-control css-0"
              data-testid="TextField"
              role="group"
            >
              <label
                class="chakra-form__label css-0"
                for="field-:r2:"
                id="field-:r2:-label"
              >
                Email
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <div
                class="css-0"
              >
                <input
                  aria-required="true"
                  autocomplete="email"
                  class="chakra-input css-0"
                  id="field-:r2:"
                  name="email"
                  placeholder="Пароль"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="chakra-form-control css-0"
              data-testid="TextField"
              role="group"
            >
              <label
                class="chakra-form__label css-0"
                for="field-:r3:"
                id="field-:r3:-label"
              >
                Пароль
                <span
                  aria-hidden="true"
                  class="chakra-form__required-indicator css-0"
                  role="presentation"
                >
                  *
                </span>
              </label>
              <div
                class="css-0"
              >
                <input
                  aria-required="true"
                  autocomplete="password"
                  class="chakra-input css-0"
                  id="field-:r3:"
                  name="password"
                  placeholder="Пароль"
                  required=""
                  type="password"
                  value=""
                />
                <button
                  aria-label=""
                  class="chakra-button css-0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    font-size="24px"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 576 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="css-0"
            >
              <button
                class="chakra-button css-0"
                type="submit"
              >
                Войти
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="css-0"
      data-testid="Header"
    >
      <div
        class="chakra-container css-0"
      >
        <div
          class="css-0"
        >
          <a
            href="/"
          >
            <h2
              class="chakra-heading css-0"
            >
              IMS-Docs
            </h2>
          </a>
          <div
            class="css-0"
          >
            <button
              aria-label="Switch to dark mode"
              class="chakra-button css-0"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </header>
    <div
      class="css-0"
      data-testid="Login"
    >
      <h2
        class="chakra-heading css-0"
      >
        Авторизация
      </h2>
      <div
        class="css-0"
        data-testid="FormComponent"
      >
        <form
          enctype="multipart/form-data"
          style="width: 100%;"
        >
          <div
            class="chakra-form-control css-0"
            data-testid="TextField"
            role="group"
          >
            <label
              class="chakra-form__label css-0"
              for="field-:r2:"
              id="field-:r2:-label"
            >
              Email
              <span
                aria-hidden="true"
                class="chakra-form__required-indicator css-0"
                role="presentation"
              >
                *
              </span>
            </label>
            <div
              class="css-0"
            >
              <input
                aria-required="true"
                autocomplete="email"
                class="chakra-input css-0"
                id="field-:r2:"
                name="email"
                placeholder="Пароль"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="chakra-form-control css-0"
            data-testid="TextField"
            role="group"
          >
            <label
              class="chakra-form__label css-0"
              for="field-:r3:"
              id="field-:r3:-label"
            >
              Пароль
              <span
                aria-hidden="true"
                class="chakra-form__required-indicator css-0"
                role="presentation"
              >
                *
              </span>
            </label>
            <div
              class="css-0"
            >
              <input
                aria-required="true"
                autocomplete="password"
                class="chakra-input css-0"
                id="field-:r3:"
                name="password"
                placeholder="Пароль"
                required=""
                type="password"
                value=""
              />
              <button
                aria-label=""
                class="chakra-button css-0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  fill="currentColor"
                  focusable="false"
                  font-size="24px"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="css-0"
          >
            <button
              class="chakra-button css-0"
              type="submit"
            >
              Войти
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
